lib/tapyrus/rpc/tapyrus_core_client.rb

```diff
@@ -62,13 +94,17 @@ def request(command, *params)
         request.content_type = 'application/json'
         request.body = data.to_json
         response = http.request(request)
-        body = response.body
-        response = Tapyrus::Ext::JsonParser.new(body.gsub(/\\u([\da-fA-F]{4})/) { [$1].pack('H*').unpack('n*').pack('U*').encode('ISO-8859-1').force_encoding('UTF-8') }).parse
-        raise response['error'].to_json if response['error']
+        raise Error.new(response) unless response.is_a? Net::HTTPOK
+        response = Tapyrus::RPC::response_body2json(response.body)
         response['result']
       end
 
     end
 
+    def response_body2json(body)
+      Tapyrus::Ext::JsonParser.new(body.gsub(/\\u([\da-fA-F]{4})/) { [$1].pack('H*').unpack('n*').pack('U*').encode('ISO-8859-1').force_encoding('UTF-8') }).parse
+    end
+
+    module_function :response_body2json
```

Comment:
Why do you need to expose it externally as a module function?

Reply:
For want to use this function both of parse success result and parse error message in Tapyrus::RPC::Error.